libs/wasm/detective/src/lib.rs

```diff
@@ -26,7 +26,7 @@ pub extern "C" fn f(ptr: *mut u8, length: usize) -> u64 {
             None,
             None,
             None,
-            WASMExitCode::WASM_EXIT_CODE_ERROR,
+            WASMExitCode::WASM_EXIT_CODE_FALSE,
```

Comment:
Didn't notice this at review - the `validate_wasm_request()` call failures should return err. I pushed another PR with updates.